services/scraper/scraper.py

```diff
@@ -2,41 +2,65 @@
 
 
 import re
-import time
 from datetime import datetime, timezone
 import requests
 from bs4 import BeautifulSoup
 from celery import shared_task
 from selenium.webdriver.common.by import By
 from mongoengine.errors import NotUniqueError
-from utils.utils import init_driver, get_date, ScrapingTask, TaskStatus
+from utils.utils import init_driver, get_date
 from database.models import ScrapedAuthor, Author, Quote, ScheduledTask
 from constants.app_constants import QUOTES_URL
+from services.scraper.utils import ScrapingTask, TaskStatus
 
 
 @shared_task(bind=True)
-def scrape_data(self, single_page: bool = False) -> dict:
-    """This function scrape quotes and authors data at a time.
+def scrape_data_scheduler(self, task_id: str) -> dict:
+    """This function scrape quotes and authors data as scheduled tasks.
 
     Args:
         single_page (bool): bool to scrape only single page
 
     Returns:
         dict: dict of no. of fetched records
     """
-    # Wait till task not saved to database
-    time.sleep(1)
+    scheduled_task = ScheduledTask.objects.get(id=task_id)
+    scheduled_task.update(celeryTask=self.request.id)
+
+    task_status = ScrapingTask(task=scheduled_task)
+
+    task_status.update(quote=TaskStatus.IN_PROGRESS)
+    try:
+        quote_status = scrape_quotes(self, single_page=False)
+        task_status.update(quote=TaskStatus.SUCCESS)
+    except Exception:
+        quote_status = self.AsyncResult(self.request.id).info
+        task_status.update(quote=TaskStatus.FAILED)
+
+    task_status.update(author=TaskStatus.IN_PROGRESS)
+    try:
+        author_status = scrape_authors(self)
+        task_status.update(author=TaskStatus.SUCCESS)
+    except Exception:
+        author_status = self.AsyncResult(self.request.id).info
+        task_status.update(author=TaskStatus.FAILED)
 
-    scheduled_task = ScheduledTask.objects.get(celeryTask=self.request.id)
-    task = ScrapingTask(task=scheduled_task)
+    return {"quotes": quote_status, "authors": author_status}
 
-    task.update(quote=TaskStatus.IN_PROGRESS)
-    quote_status = scrape_quotes(self, single_page)
 
-    task.update(quote=TaskStatus.SUCCESS, author=TaskStatus.IN_PROGRESS)
+@shared_task(bind=True)
+def scrape_data(self, single_page: bool = False) -> dict:
+    """This function scrape quotes and authors data at a time.
+
+    Args:
+        single_page (bool): bool to scrape only single page
+
+    Returns:
+        dict: dict of no. of fetched records
+    """
+    quote_status = scrape_quotes(self, single_page)
     author_status = scrape_authors(self)
```

Comment:
You might need to handle error here! or have to handled them inside these functions?